### DISCUSSION TITLE
Notifications keep un-muting on desktop client when linked to signal-cli?

### DISCUSSION THREAD

**Discussion by codywohlers** [AsamK/signal-cli#1537] [Q&A] [answered]
On linux, I have signal-cli running in dameon mode and the official signal-desktop client linked to signal-cli.  I mute the notifications (set to always) for each chat in the desktop client but they later get set back to unmuted automatically. 

Is this a known issue or should i report it?  It didn't happen previously when Signal Android on my phone was the main device.

**Comment by AsamK** [comment]
Not a known issue, also it works for me.
Can you provide verbose log of signal-cli when this happens?

**Reply by codywohlers** [reply]
Thanks for confirming it should work.  

tldr; i think it might be this:

```
2024-05-26T20:14:41.860-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - Remote type 0 did not match local type 6 for h8******************Vg!
2024-05-26T20:14:41.861-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - Remote type 0 did not match local type 6 for Ss******************jA!
2024-05-26T20:14:41.872-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - ID Difference :: IdDifferenceResult[remoteOnlyIds=[StorageId{type=3, raw=vIhieXXXXXXXXXXXXXRAjA}], localOnlyIds=[StorageId{type=3, raw=ss1kXXXXXXXXXXXXIxUF7A}], hasTypeMismatches=true]
2024-05-26T20:14:41.886-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - We have something to write remotely.
2024-05-26T20:14:41.886-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - WriteOperationResult :: ManifestVersion: 10957, Total Keys: 1522, Inserts: 1, Deletes: 1
2024-05-26T20:14:42.170-0600 [pool-2-thread-1] DEBUG o.a.s.manager.helper.StorageHelper - Saved new manifest. Now at version: 10957
```




Full story:

Chats are unmuting in signal-desktop.

signal-cli --version: 0.13.3
signal-desktop version: 7.10.0 production

I complied signal-cli myself with libc=2.31 for ubuntu20.04, which also involved compiling libsignal_jni.so and replacing it in libsignal-client-0.44.0.jar


The only thing I noticed in the log *without* verbose is:

`WARN  StorageHelper - Found 12 records that were deleted remotely but only marked unregistered locally. Removed those from local store.`

- the message happens multiple times and the number of records varies on each.


When I linked signal-desktop originally I still had all the old chats in signal-desktop.  So I tried clearing them and re-linking:

- deleted all account data from signal-desktop
- removed linked device from signal-cli (signal-cli removeDevice)
- re-linked signal-desktop to signal-cli (signal-cli addDevice)

same behavior, some chats would unmute when receiving messages.



So I wanted to see if it was a signal-cli or signal-desktop issue so then I tried:

1. stop signal-cli daemon
2. on signal-desktop mute all chats
3. receive several messages over a few hours in signal-desktop

now chats stay muted in signal-desktop when receiving new messages!

3. run signal-cli receive (and wait for completion) (large output so not included here)

some chats in signal-desktop get unmuted during receive!  I'll have to try and catch that output when it happens.

4. mute chats again on signal-desktop
5. run signal-cli receive again (no verbose) (and wait for completion)

```
Envelope from: “(me)” +1XXXXXXXX82 (device: 2) to +1XXXXXXXX82
Timestamp: 1716766482405 (2024-05-26T23:34:42.405Z)
Server timestamps: received: 1716766482517 (2024-05-26T23:34:42.517Z) delivered: 1716766491823 (2024-05-26T23:34:51.823Z)
Received a sync message
```

- chat stays muted (but no messages received during it)


- seems to unmute in signal-desktop when messages received (i'm guessing a sync or something is done with signal-cli?)





Log of chat being NOT umuted in signal-desktop when `signal-cli --verbose --scrub-log receive` is ran (correct behavior):
(and some additional manual scrubbing "XXXXX")

https://gist.github.com/codywohlers/b0781cb90d2fa83eb01a19566d72dd62#file-signal-cli-log-normal-txt

And here's a 3 chats being umuted in signal-desktop when `signal-cli --verbose --scrub-log receive` is ran:

https://gist.github.com/codywohlers/b00c55e102e38b6236a2e139971f87b4#file-signal-cli-log-unmutes-chats-txt



3 chats were unmuted and the only difference in the verbose log between each section above is 3 instances of:

```
[pool-2-thread-X] DEBUG o.a.s.manager.helper.StorageHelper - Remote type 0 did not match local type 6 for h8******************Vg
[pool-2-thread-X] DEBUG o.a.s.manager.helper.StorageHelper - Remote type 0 did not match local type 6 for Ss******************jA!
[pool-2-thread-X] DEBUG o.a.s.manager.helper.StorageHelper - ID Difference :: IdDifferenceResult[remoteOnlyIds=[], localOnlyIds=[], hasTypeMismatches=true]
[pool-2-thread-X] DEBUG o.a.s.manager.helper.StorageHelper - No remote writes needed. Still at version: 10958
[pool-2-thread-X] DEBUG o.a.s.manager.helper.StorageHelper - Done syncing data with remote storage

```

**Comment by k-bx** [comment]
I've started this conversation on Signal Android, didn't have a clear confirmation on who to blame but yes, it definitely does happen for heavy users like myself https://github.com/signalapp/Signal-Android/issues/13535

**Comment by codywohlers** [comment]
WORKAROUND

My workaround for now is to not run signal-cli until I need to link a device and just use the linked devices.  The only issues with this I see are (I think):
- after 7 days I'll loose my PIN lock for the account? (because my primary device hasn't connected?)
- after 30? 90? days I'll loose messages from the queue for the primary device.
but the account won't expire right?  Or do i have to connect with my primary device every so often even though I connect daily with linked devices?

**Comment by codywohlers** [comment, marked as answer]
SOLVED

This was an issue in Signal-Desktop and was fixed in version `7.40.1`
https://github.com/signalapp/Signal-Android/issues/13535#issuecomment-2629492140
https://github.com/signalapp/Signal-iOS/issues/5956#issuecomment-2629275427

And to answer my question about idle signal accounts, the best I can deduce is:
You must connect with a _primary or secondary device_ at least every **120 days** (https://old.reddit.com/r/signal/comments/1e3ecq7/deleted_by_user/ldabnhg/)
And you must download messages on your **primary device at least once a year** (https://support.signal.org/hc/en-us/articles/8997185514138-Re-connect-your-primary-device-to-continue-using-Signal-Desktop)
